apps/astarte_realm_management/lib/astarte_realm_management/queries.ex

```diff
@@ -920,52 +920,57 @@ defmodule Astarte.RealmManagement.Queries do
-        client,
+        realm_name,
         object_id,
         object_type,
         parent_trigger_id,
         simple_trigger_id,
         simple_trigger,
         trigger_target
       ) do
+    keyspace = Realm.keyspace_name(realm_name)
+
     insert_simple_trigger_statement = """
-    INSERT INTO simple_triggers
+    INSERT INTO #{keyspace}.simple_triggers
     (object_id, object_type, parent_trigger_id, simple_trigger_id, trigger_data, trigger_target)
-    VALUES (:object_id, :object_type, :parent_trigger_id, :simple_trigger_id, :simple_trigger_data, :trigger_target_data);
+    VALUES (?, ?, ?, ?, ?, ?);
     """
 
-    insert_simple_trigger_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(insert_simple_trigger_statement)
-      |> DatabaseQuery.put(:object_id, object_id)
-      |> DatabaseQuery.put(:object_type, object_type)
-      |> DatabaseQuery.put(:parent_trigger_id, parent_trigger_id)
-      |> DatabaseQuery.put(:simple_trigger_id, simple_trigger_id)
-      |> DatabaseQuery.put(:simple_trigger_data, SimpleTriggerContainer.encode(simple_trigger))
-      |> DatabaseQuery.put(:trigger_target_data, TriggerTargetContainer.encode(trigger_target))
-
-    astarte_ref = %AstarteReference{
-      object_type: object_type,
-      object_uuid: object_id
-    }
-
-    insert_simple_trigger_by_uuid_statement =
-      "INSERT INTO kv_store (group, key, value) VALUES ('simple-triggers-by-uuid', :simple_trigger_id, :astarte_ref);"
+    insert_params = [
+      object_id,
+      object_type,
+      parent_trigger_id,
+      simple_trigger_id,
+      SimpleTriggerContainer.encode(simple_trigger),
+      TriggerTargetContainer.encode(trigger_target)
+    ]
 
-    insert_simple_trigger_by_uuid_query =
-      DatabaseQuery.new()
-      |> DatabaseQuery.statement(insert_simple_trigger_by_uuid_statement)
-      |> DatabaseQuery.put(:simple_trigger_id, :uuid.uuid_to_string(simple_trigger_id))
-      |> DatabaseQuery.put(:astarte_ref, AstarteReference.encode(astarte_ref))
+    astarte_ref =
+      %AstarteReference{
+        object_type: object_type,
+        object_uuid: object_id
+      }
+      |> AstarteReference.encode()
+
+    simple_trigger_id =
+      simple_trigger_id
+      |> :uuid.uuid_to_string()
+      |> to_string()
+
+    kv_insert =
+      %{
+        group: "simple-triggers-by-uuid",
+        key: simple_trigger_id,
+        value: astarte_ref
+      }
+      |> KvStore.insert_sql(keyspace)
 
-    with {:ok, _res} <- DatabaseQuery.call(client, insert_simple_trigger_query),
-         {:ok, _res} <- DatabaseQuery.call(client, insert_simple_trigger_by_uuid_query) do
-      :ok
-    else
-      not_ok ->
-        _ = Logger.warning("Database error: #{inspect(not_ok)}.", tag: "db_error")
-        {:error, :cannot_install_simple_trigger}
-    end
+    Exandra.execute_batch(
+      Repo,
+      %Exandra.Batch{
+        queries: [{insert_simple_trigger_statement, insert_params}, kv_insert]
+      }
+    )
```

Comment:
:eyes: you're changing the ordering of queries here: before it was `insert_simple_trigger_query` and then `insert_simple_trigger_by_uuid_query`, now it's the opposite

Reply:
Also, is it really necessary to use a batch for a single insert?